src/braket/pennylane_plugin/translation.py

```diff
@@ -57,19 +40,177 @@ def supported_operations() -> FrozenSet[str]:
     Returns:
         FrozenSet[str]: The names of the supported operations
     """
-    return frozenset(_OPERATION_MAP.keys())
+    return frozenset(op.__name__ for op in translate_operation.registry)
 
 
-def translate_operation(operation: Operation) -> Type[Gate]:
-    """Translates a PennyLane ``Operation`` into the corresponding Braket gate.
+@singledispatch
+def translate_operation(operation: Operation, parameters) -> Gate:
+    """Translates a PennyLane ``Operation`` into the corresponding Braket ``Gate``.
 
     Args:
-        operation (Operation): The Pennylane ``Operation`` to translate
+        operation (Operation): The PennyLane ``Operation`` to translate
+        parameters: The parameters of the operation
 
     Returns:
-        Type[Gate]: The `Gate` class corresponding to the given operation
+        Gate: The Braket gate corresponding to the given operation
     """
-    return _OPERATION_MAP[operation.name]
+    raise NotImplementedError(
+        f"Braket PennyLane plugin does not support operation {operation.name}."
+    )
```

Comment:
Having `singledispatch` on a public method looks really ugly in the docs. I'd suggest having a private proxy method `_translate_operation` and decorate that one with `singledispatch`, and just have `translate_operation` (public) call the private method.

Reply:
agreed